src/macaron/slsa_analyzer/analyzer.py

```diff
@@ -167,18 +171,46 @@ def run(self, user_config: dict, skip_deps: bool = False) -> int:
             find_ctx = report.find_ctx(dup_record.pre_config.get_value("path"))
             dup_record.context = find_ctx
 
-        # Store the analysis result into the SQLite database.
-        for ctx in report.get_ctxs():
-            self.store_result_to_db(ctx)
+        # Store the analysis result(s) into the SQLite database.
+        self.db_man.create_tables()
+
+        analysis = store_analysis_to_db(self.db_man, main_record)
+
+        for record in report.get_records():
+            if not record.status == SCMStatus.DUPLICATED_SCM:
+                if record.context:
+                    store_analyze_context_to_db(self.TABLE_NAME, self.db_man, analysis, record.context)
+
+        # Store dependency relations
+        for parent, child in report.get_dependencies():
+            dependency = RepositoryDependency(
```

Comment:
This is now a Core API table as of https://github.com/oracle-samples/macaron/pull/46/commits/4e61112595339f7871796e707f604cc1ff1b5711, but will wait to upgrade to SQLAlchemy 2 to add the list since the 1.4 API is a lot more convoluted for that `relationship`